src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt

```diff
@@ -24,29 +22,123 @@ class ThemePatch : ResourcePatch {
         val darkThemeBackgroundColor = darkThemeBackgroundColor!!
         val lightThemeBackgroundColor = lightThemeBackgroundColor!!
 
-        context.xmlEditor["res/values/colors.xml"].use { editor ->
-            val resourcesNode = editor.file.getElementsByTagName("resources").item(0) as Element
+        val valuesPaths: List<String> = listOf(
+            "values", // Android 11 (and below)
+            "values-v31" // Android 12 (and above)
+        )
```

Comment:
These should be two separate constants to prevent odd code [such as this](https://github.com/revanced/revanced-patches/pull/1315/files#diff-5a51aa8d702357ed352218e03738bda901cd9181c815090f201b6060b753dddeR31). You would create an array out of the constants later below [here](https://github.com/revanced/revanced-patches/pull/1315/files#diff-5a51aa8d702357ed352218e03738bda901cd9181c815090f201b6060b753dddeR84).